tldraw/tldraw
# feat(core): add resize observer and improve embedded components

This PR includes several changes that improve the TLDraw and Renderer components.

- Adds an update and size property to the Inputs class to support correct offsets
- Adds a ResizeObserver to update the offsets when the component's dimensions change
- Improves window resize handling
- Bumps pan and wheel gesture library to the latest version of `@useGesture/react`
- Scopes the CSS variables to the component's container element, allowing for multiple TLDraw instances
- Likewise with the event handlers for scroll, wheel and pan
- Improved pointer / touch event handling, preventing wrong behaviour when pinching during a session
- Improves resize handling
- Improves touch sizes on mobile

### Change type


### Test plan

1. Open the embedded example and verify that resizing the container updates the canvas offsets correctly.
2. Verify that multiple TLDraw instances on the same page do not conflict with each other.


### Release notes

- Improved support for embedded components and multiple instances via ResizeObserver and scoped CSS.
